tests/test_picai/test_case.py

```diff
@@ -0,0 +1,63 @@
+import os
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import SimpleITK as sitk
+
+from research.picai.preprocessing import PicaiCase, PreprocessingSettings
+
+
+def test_read_and_write_picai_case() -> None:
+    X, Y, Z = 32, 64, 128
+    with tempfile.TemporaryDirectory() as results_dir:
```

Comment:
I think this is fine, but pytest actually offers a temporary file system that it cleans up itself for this type of thing. You can see it in `tests/checkpointing/test_save_load.py`. 
```
def test_save_and_load_best_loss_checkpoint(tmp_path: Path) -> None:
    # Temporary path to write pkl to, will be cleaned up at the end of the test.
    checkpoint_dir = tmp_path.joinpath("resources")
    checkpoint_dir.mkdir()
    ...
```
I think it's a little cleaner. At the very least it saves some indents 😂 , but if there is a reason you like this better, I'm happy to hear it 🙂